src/components/header/FandomGlobalHeaderDesktop.es6

```diff
@@ -0,0 +1,291 @@
+import Headroom from 'headroom.js';
+import headerTemplate from './templates/desktop/header.handlebars';
+import anonUserMenu from './templates/desktop/anon.handlebars';
+import userMenu from './templates/desktop/user.handlebars';
+import userLink from './templates/desktop/userLink.handlebars';
+import userMenuLogout from './templates/desktop/userMenuLogout.handlebars';
+import { ATTRIBUTES } from '../../helpers/AttributeHelper.es6';
+import SvgHelper from '../../helpers/svg/SvgHelper.es6';
+import PopupHelper  from '../../helpers/PopupHelper.es6';
+import getStrings from '../../getStrings.es6';
+import { EVENTS } from './events.es6';
+import { fromNavResponse, validateUserData } from './userData.es6';
+import { request, requestNavInfo } from './services.es6';
+
+const headroomElementSelector = 'header.wds-global-navigation';
+
+// desktop
+const CSS_CLASSES = {
+    HEADROOM_PINNED: 'headroom--pinned',
+    HEADROOM_UNPINNED: 'headroom--unpinned',
+    SEARCH_ACTIVE: 'wds-search-is-active',
+    USER_LOGGED_IN: 'wds-user-is-logged-in'
+};
+
+// desktop (will need mobile separately)
+const HEADROOM_OPTIONS = {
+    offset: 0,
+    tolerance: {
+        up: 5,
+        down: 0
+    },
+    classes: {
+        pinned: CSS_CLASSES.HEADROOM_PINNED,
+        unpinned: CSS_CLASSES.HEADROOM_UNPINNED
+    }
+};
+
+export default class FandomGlobalHeader {
+    constructor(el, parent) {
+        this.el = el;
+        this.parent = parent;
+        this.atts = this.parent.data.attributes;
+        this.popup = new PopupHelper((e) => this._onMessage(e));
+        this.strings = getStrings(this.atts['lang-code']);
+        this.headroom = null;
+        this.svgs = new SvgHelper(this.el);
+    }
+
+    init() {
+        this._draw();
+        this.headroom = new Headroom(
+            this.el.querySelector(headroomElementSelector),
+            this._buildHeadroomOptions()
+        );
+
+        this._updateUserState();
+        this._updateNavLinks();
+
+        this._bindSearchActions();
+
+        this.parent.onEvent(EVENTS.USER_DATA_REFRESHED, () => this._updateUserState());
+
+        return this;
+    }
+
+    isVisible() {
```

Comment:
we call this from the FC in `StickyController`. If we're going to remove it from here we need to make sure to account for the height diff somehow (maybe have it listen to the pin/unpinned event?)